### PR TITLE
feat(new-trace): Route to traceView if traceSlug from query is not null.

### PR DESCRIPTION
**Pull request by Abdkhan14**
Initially we were only routing to the trace from when the event_id was associated to a transaction.

### CODE REVIEW

File: static/app/views/discover/table/tableView.tsx [hunk 1]
```diff
@@ -189,13 +189,23 @@ function TableView(props: TableViewProps) {
...
           throw new Error(
             'Transaction event should always have a trace associated with it.'
           );
         }
 
+        const project = dataRow.project || dataRow['project.name'];
+
+        target = {
+          // NOTE: This uses a legacy redirect for project event to the issue group event link
+          // This only works with dev-server or production.
```
Comment (narsaynorath) [inline static/app/views/discover/table/tableView.tsx:204]: Just tested this out a bit on my `dev-ui` and it seems to work. Is this comment still necessary? I didn't get any weird infinite redirects or anything.

Comment (narsaynorath) [reply static/app/views/discover/table/tableView.tsx:204]: Also should we wrap the `to={target}` in the styled link with `normalizeUrl()`?

**Review by narsaynorath — APPROVED** [review]
Tested it locally on the org that raised the issue

File: static/app/views/discover/table/tableView.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/discover/table/tableView.tsx:204]: Good point, I wrapped the path from the `trace:null` case with `normalizeUrl()` the other path already uses it.

File: static/app/views/discover/table/tableView.tsx [hunk 2]
```diff
@@ -190,15 +190,14 @@ function TableView(props: TableViewProps) {
 
       let target;
 
-      if (dataRow.trace === null) {
+      if (dataRow.trace == null) {
```
Comment (JonasBa) [inline static/app/views/discover/table/tableView.tsx:193]: This is misleading and can be really hard to catch when looking for bugs as it is actually checking datarow.trace === null || datarow.trace === undefined. You should either use !datarow.trace or do an explicit check for both value types

**Pull request merged**